recipes/datastax.rb

```diff
@@ -146,11 +152,21 @@
     options node['cassandra']['yum']['options']
   end
 
+  if node['cassandra']['use_systemd'] == true
+    file '/etc/init.d/' + node['cassandra']['service_name'] do
+      action :delete
+    end
+  end
+
   # applying fix for java search directories, on java 8 it needs to be update
   # including the new directories
   ruby_block 'set_jvm_search_dirs_on_java_8' do
     block do
-      init_path = ::File.join('/etc/init.d/', node['cassandra']['service_name'])
+      if node['cassandra']['use_systemd'] == false
+        init_path = ::File.join('/etc/init.d/', node['cassandra']['service_name'])
```

Comment:
Since you assign the same variable in both branches, it is very common to rely on the fact that `if` is an expression in Ruby:

``` ruby
init_value = if …
                      …
                    else
                      …
                    end
```